packages/core/src/external/saml/xcpd/iti55-envelope.ts

```diff
@@ -281,12 +264,24 @@ export function createITI5SoapEnvelope({
   return xmlContent;
 }
 
-export function createAndSignXCPDRequest(
-  bodyData: XCPDBodyData,
+export function createAndSignBulkXCPDRequests(
+  bulkBodyData: OutboundPatientDiscoveryReq,
   publicCert: string,
-  privateKey: string
-): string {
-  const xmlString = createITI5SoapEnvelope({ bodyData, publicCert });
-  const fullySignedSaml = signFullSaml({ xmlString, publicCert, privateKey });
-  return fullySignedSaml;
+  privateKey: string,
+  privateKeyPassword: string
+): BulkSignedXCPD[] {
+  const signedRequests: BulkSignedXCPD[] = [];
+
+  for (const gateway of bulkBodyData.gateways) {
+    const bodyData: OutboundPatientDiscoveryReq = {
+      ...bulkBodyData,
+      gateways: [gateway],
+    };
```

Comment:
What we are doing here is taking a request with a lot of gateways, and converting it into many requests each with one gateway